rest-server/auth-server/src/main/scala/com/actionml/authserver/dal/mongo/ClientsDaoImpl.scala

```diff
@@ -27,11 +27,12 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class ClientsDaoImpl(implicit inj: Injector) extends ClientsDao with MongoSupport with Injectable {
   implicit private val ec = inject[ExecutionContext]
-  private val clients = collection[Client]("clients")
+  private lazy val clients = collection[Client]("clients")
 
-  override def find(id: String): Future[Client] = {
+  override def find(id: String): Future[Option[Client]] = {
     clients.find(equal("id", id))
       .first
       .toFuture
+      .map(Option.apply)
   }
 }
```

Comment:
I like the DAO pattern you are using. I use Salat for generic DAOs but they cause Mongo queries to be used everywhere. This is much cleaner.